src/nodes/conv.rs

```diff
@@ -50,6 +51,12 @@ fn system_time_to_timespec(path: &Path, name: &str, time: &io::Result<SystemTime
     }
 }
 
+/// Converts a `time::Timespec` object into a `sys::time::TimeVal`.
+pub fn timespec_to_timeval(spec: Timespec) -> sys::time::TimeVal {
```

Comment:
Ugh, this function makes me cringe. 😢 But I think there's no better way. It's annoying that TimeVal is (needlessly) opaque and doesn't have a sec+nsec constructor.

Reply:
I've added a TODO to get this addressed upstream.

(There are way too many time representations involved here :-/ )